### PR TITLE
test(cursor): ensure that we properly kill cursors

### PR DESCRIPTION
**Pull request by daprahamian**
Added a test to make sure we properly kill cursors

Fixes NODE-1214

### CODE REVIEW

File: test/functional/cursor_tests.js [hunk 1]
```diff
@@ -4163,6 +4163,95 @@ describe('Cursor', function() {
...
+            const db = client.db(configuration.db);
+            const collection = db.collection('cursorkilltest1');
+
+            // Insert 1000 documents
+            return collection.insert(docs).then(() => {
+              // Generate cursor for find operation
+              const cursor = collection.find({});
+
+              // Iterate cursor past first element
+              return cursor
```
Comment (mbroadst) [inline test/functional/cursor_tests.js]: I get keeping the promise chain one level deeper so you can maintain access to the `client` variable, but it looks like right about here we can keep the chain just at the level of the `collection.insert(docs)` above - starting to look a little callback hellish :)

File: test/functional/cursor_tests.js [hunk 2]
```diff
@@ -4163,6 +4163,95 @@ describe('Cursor', function() {
...
+          a: i
+        });
+      }
+
+      const configuration = this.configuration;
+      const client = configuration.newClient(configuration.writeConcernMax(), { poolSize: 1 });
+      let cleanup = () => {};
+      let caughtError = undefined;
+
+      return (
```
Comment (mbroadst) [inline test/functional/cursor_tests.js]: what's the intent in wrapping all of it in parentheses?

File: test/functional/cursor_tests.js [hunk 3]
```diff
@@ -4163,6 +4163,95 @@ describe('Cursor', function() {
...
+                      .to.be.an('object')
+                      .and.to.haveOwnProperty('_bsontype', 'Long');
+                    id = id.toNumber();
+                  }
+
+                  expect(id).to.not.equal(0);
+
+                  // Kill cursor
+                  return new Promise((resolve, reject) => {
+                    cursor.kill((err, response) => {
```
Comment (mbroadst) [inline test/functional/cursor_tests.js]: nitpick, but can't we make this very short? `cursor.kill((err, r) => err ? reject(err) : resolve(r))`

File: test/functional/cursor_tests.js [hunk 4]
```diff
@@ -4163,6 +4163,95 @@ describe('Cursor', function() {
...
+              const cursor = collection.find({});
+
+              // Iterate cursor past first element
+              return cursor
+                .next()
+                .then(() => cursor.next())
+                .then(() => {
+                  // Confirm that cursorId is non-zero
+                  let id = cursor.cursorState.cursorId;
+                  if (typeof id !== 'number') {
```
Comment (mbroadst) [inline test/functional/cursor_tests.js]: did you find a case where the id is not a `Long`? Seems like maybe we can cut this extra code?

File: test/functional/cursor_tests.js (hunk 1 above)
Comment (mbroadst) [reply test/functional/cursor_tests.js]: oh I take it back, you're using the `cursor` as well, disregard

File: test/functional/cursor_tests.js (hunk 2 above)
Comment (daprahamian) [reply test/functional/cursor_tests.js]: prettier did that automatically :(

**Pull request merged**